src/components/Loading/Loading.react.js

```diff
@@ -0,0 +1,88 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import GraphSpinner from './spinners/GraphSpinner.jsx';
+import DefaultSpinner from './spinners/DefaultSpinner.jsx';
+import CubeSpinner from './spinners/CubeSpinner.jsx';
+import CircleSpinner from './spinners/CircleSpinner.jsx';
+import DotSpinner from './spinners/DotSpinner.jsx';
+
+/**
+ * A Loading component that wraps any other component and displays a spinner until the wrapped component has rendered.
+ */
+export default class Loading extends Component {
+    render() {
+        const { loading_state, color, fullscreen, debug } = this.props;
+        if (loading_state&& loading_state.is_loading) {
+            switch (this.props.type) {
+                case 'graph':
+                    return <GraphSpinner status={loading_state} color={color} debug={debug} fullscreen={fullscreen}/>;
+                case 'cube':
+                    return <CubeSpinner status={loading_state} color={color} debug={debug} fullscreen={fullscreen} />;
+                case 'circle':
+                    return <CircleSpinner status={loading_state} color={color} debug={debug} fullscreen={fullscreen} />;
+                case 'dot':
+                    return <DotSpinner status={loading_state} color={color} debug={debug} fullscreen={fullscreen} />;
+                default:
+                    return <DefaultSpinner status={loading_state} color={color} debug={debug} fullscreen={fullscreen} />;
+            }
+        }
```

Comment:
Since all spinners have the same props, you could simplify this
```
function getSpinner(type) {
    switch (type) {
        case 'graph':
            return GraphSpinner;
        case 'cube':
            return CubeSpinner;
        case 'circle':
            return CircleSpinner;
        case 'dot':
            return DotSpinner;
        default:
            return DefaultSpinner;
    }
}
```

```
const Spinner = getSpinner(this.props.type); // Capitalized for React's sake!

return <Spinner
    status={loading_state}
    color={color}
    debug={debug}
    fullscreen={fullscreen}
/>;
```